module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,27 @@
+#include "compression.hpp"
+
+#include <algorithm>
+#include <utility>
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(const std::array<std::array<uint8_t, 240>, 160>& image) {
+    std::vector<std::pair<uint8_t, uint8_t>> compressed{};
+
+    for (const auto& row : image) {
+        for (auto it = row.begin(); it != row.end();) {
+            auto findIt = std::adjacent_find(it, row.end(), std::not_equal_to<int>{});
+            if (findIt != row.end()) {
+                findIt++;
+            }
+            compressed.emplace_back(*it, std::distance(it, findIt));
+            it = findIt;
+        }
+    }
```

Comment:
Pierwszą pętlę można zastąpić?
 std::for_each(image.begin(), image.end(), drugaPetla);

Drugą pętlę można zastąpić?
std::for_each_n(row.begin(),  std::prev(row.end(), 1), działanie);
użycie prev = #include <iterator>

Reply:
Mateusz napisał na discordzie, że jak nie mamy pojęcia jak zastąpić pętle to lepiej je zostawi i powie coś więcej dzisiaj na wykładzie